pkg/vscode/v1/Makefile

```diff
@@ -0,0 +1,37 @@
+# GENERATE_VSCODE_SETTINGS can be set to false so that the user can opt
+# out of generating the VSCode settings file.
+GENERATE_VSCODE_SETTINGS ?= true
+
+# VSCODE_SETTINGS_FILE is the VSCode workspace settings file that contains
+# workspace-specific settings that override user-defined settings and their
+# defaults.
+# More about VSCODE workspace settings can be found at:
+# https://code.visualstudio.com/docs/getstarted/settings#_workspace-settings
+VSCODE_SETTINGS_FILE = $(MF_PROJECT_ROOT)/.vscode/settings.json
+
+ifeq ($(GENERATE_VSCODE_SETTINGS),true)
+GENERATED_FILES += $(VSCODE_SETTINGS_FILE)
```

Comment:
One potential problem I can see with this approach is that previous user's VSCode settings in a repository can be wiped out if `make regenerate` is run.

Reply:
Yeah, it's definitely bit hacky to try to make it so they can have their own and _also_ have it managed by Makefiles. 

That might be a non issue though, the fact that it's opt-in is good enough for me for now :)